### ISSUE TITLE
[SSD/PyTorch] Negative Hard Mining counts positive boxes twice

### ISSUE THREAD

**Issue by pulkitmadaan-wiai** [NVIDIA/DeepLearningExamples#948]
Related to **SSD/PyTorch** 

**Describe the bug**
SSD/sec/model.py [[Line 190](https://github.com/NVIDIA/DeepLearningExamples/blob/ff6f7c6532e50f0d6b7268b3e3ee66083dd260c7/PyTorch/Detection/SSD/src/model.py#L190)] : `closs = (con*(mask.float() + neg_mask.float())).sum(dim=1)`
Typecast `float()` should come after addition, otherwise loss of positive boxes get counted twice. This makes the ratio of negative to positive 3:2, when in the paper it is 3:1.

**To Reproduce**
Logical Error. Will always happen.

**Expected behavior**
`  closs = (con*((mask + neg_mask).float()).sum(dim=1)`

**Environment**
Please provide at least:
* Container version: All
* GPUs in the system: Every
* CUDA driver version: All

**Comment by ahmadki** [comment]
Hi  @pulkitmadaan-wiai 

The issue you are describing can only occure when the number of positive detections is larger than `1/4` of the bboxes the network produces (`pos_num>8732/4`). In that case, both `neg_mask[i]` and `mask[i]` will be `True` for some indices `i`.

I don't think this is an issue (at least not with coco or voc datasets), something would have already gone terribly wrong for the network to produce over 2183 positive detection in a single image.

However, I do agree that 

```
closs = (con*((mask + neg_mask).float())).sum(dim=1)
```

is more clear and covers all edge cases (Although less performant) and the implementation will be updated in future releases. 

Thanks for the feedback
